app/models/orchestration_stack.rb

```diff
@@ -93,6 +97,10 @@ def stdout(format = nil)
     format.nil? ? try(:raw_stdout) : try(:raw_stdout, format)
   end
 
+  def set_tenant_from_group
```

Comment:
Does the `OwnershipMixin` already do this for you?
(will this code work if you remove this validation and method?)

The logic behind the scenes for `set_tenant_from_group` is quite complex. Otherwise we end up running quite a few queries for code that is not altering the group.

Reply:
Nope. It won't work if you remove the validation and method. 

Also, yeah, they are beautiful tests, aren't they? Stolen directly from you.